### PR TITLE
Modify precommit config yaml and change name of typing step to Typing + pylint.

### LINKED ISSUE #49526: CI: only run pylint in CI
pylint is helping unearth some issues, but it's quite slow, especially for local development

as brought up on Slack by @jorisvandenbossche , let's just run it in CI instead

Task would be:

1. in

https://github.com/pandas-dev/pandas/blob/57d8d3a7cc2c4afc8746bf774b5062fa70c0f5fd/.pre-commit-config.yaml#L62-L65

add
```
        language: system
        pass_filenames: false
        stages: [manual]
```
2. modify the name of https://github.com/pandas-dev/pandas/blob/e02133c6323ddfd886784d6a9d3a4e5c07c99557/.github/workflows/code-checks.yml#L89-L93 to "Typing + pylint"

### PR DESCRIPTION
**Pull request by sudhansubhushanmishra**
Modify the pre-commit config.yml file and change name of step in code-checks.yml. Possible attempt to resolve #49526 
Please provide feedback an I'd try to incorporate those changes.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -63,6 +63,9 @@ repos:
     rev: v2.15.5
     hooks:
     -   id: pylint
+    -   language: system
+    -   pass_filenames: false
+    -   stages: [manual]
```
Comment (MarcoGorelli) [inline .pre-commit-config.yaml:68]: this isn't quite right, check the mypy hook for an example of the syntax

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
Comment (the-lazy-learner) [reply .pre-commit-config.yaml]: I'm sorry, are the changes ok now?

Comment (MarcoGorelli) [reply .pre-commit-config.yaml]: should be fine, thanks!

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -63,6 +63,8 @@ repos:
     rev: v2.15.5
     hooks:
     -   id: pylint
+        language: system
+        stages: [manual]
```
Comment (akx) [inline .pre-commit-config.yaml:67]: What are the implications of these changes?

**Review by akx — COMMENTED** [review]
Beyond the inline comment, it's not quite clear why more pylint codes need to be disabled..?

**Review by MarcoGorelli — APPROVED** [review]
Thanks @the-lazy-learner ! I just merged upstream/main, added `pylint` to the environment file, and added a couple more checks to the ignore list as `pylint` also has some dynamic checks which don't run when it's just a pre-commit hook, hope that's OK

**Comment by MarcoGorelli** [comment]
merging as the affected checks all pass

File: .pre-commit-config.yaml (hunk 2 above)
Comment (MarcoGorelli) [reply .pre-commit-config.yaml:67]: it'll run like the `mypy` one, only in CI

File: environment.yml [hunk 3]
```diff
@@ -89,6 +89,7 @@ dependencies:
   - flake8-bugbear=22.7.1 # used by flake8, find likely bugs
   - isort>=5.2.1  # check that imports are in the right order
   - mypy=0.981
+  - pylint=2.15.5
```
Comment (jorisvandenbossche) [inline environment.yml:92]: @MarcoGorelli is there a reason to add pylint to the env file? The check is still run through pre-commit, which will make install it in its own env?

(although that's also the case for black et al, and those are also here, so it's all OK, just wondering)

**Comment by jorisvandenbossche** [comment]
BTW, there is an actual failure:

```
 pylint...................................................................Failed
- hook id: pylint
- exit code: 8

************* Module pandas.core.dtypes.concat
pandas/core/dtypes/concat.py:284:8: R1720: Unnecessary "elif" after "raise", remove the leading "el" from "elif" (no-else-raise)

```

File: environment.yml (hunk 3 above)
Comment (MarcoGorelli) [reply environment.yml:92]: `pylint` can do some extra inference if it has everything built (hence a few extra checks that need to be turned off in this PR) - if it's only running in CI, then we might as well use the environment where pandas has been built (like we do for `mypy`, that one also has `language: system`)

**Comment by MarcoGorelli** [comment]
> BTW, there is an actual failure:
> 
> ```
>  pylint...................................................................Failed
> - hook id: pylint
> - exit code: 8
> 
> ************* Module pandas.core.dtypes.concat
> pandas/core/dtypes/concat.py:284:8: R1720: Unnecessary "elif" after "raise", remove the leading "el" from "elif" (no-else-raise)
> ```

Yup, fixed in https://github.com/pandas-dev/pandas/pull/49537

File: environment.yml (hunk 3 above)
Comment (jorisvandenbossche) [reply environment.yml:92]: Ah, I missed the `language: system` . Thanks for the explanation!

Comment (MarcoGorelli) [reply environment.yml:92]: 🤔 though `language: system` is less reliable, so maybe this should still run in its own little env...yeah let's do that, then the risk of it annoying contributors / maintainers is reduced

**Comment by MarcoGorelli** [comment]
A couple of us have had issues merging this one, not sure if the issue is on GitHub's side, but I've opened https://github.com/pandas-dev/pandas/pull/49538 to take this forwards (and have kept @the-lazy-learner as the commit author)

**Pull request closed without merging**